docs/hoon/library/2b.md

```diff
@@ -4,28 +4,132 @@ navhome: /docs/
 
 
 # 2b: list logic
-Reverse
+### `++fand`
+
+All indices in `list`
+
+Produces the indices of all occurrences of `list` `nedl` in
+`list` `hstk` as a `list` of atoms.
+
+#### Accepts
+
+`nedl` is a list.
+
+`hstk` is a list.
+
+#### Produces
+
+A `list`.
+
+#### Source
+
+    ++  fand                                                ::  all indices
+      ~/  %fand
+      |=  [nedl=(list) hstk=(list)]
+      =|  i=@ud
+      =|  fnd=(list @ud)
+      |-  ^+  fnd
+      =+  [n=nedl h=hstk]
+      |-
+      ?:  |(?=($~ n) ?=($~ h))
+        (flop fnd)
+      ?:  =(i.n i.h)
+        ?~  t.n
+          ^$(i +(i), hstk +.hstk, fnd [i fnd])
+        $(n t.n, h t.h)
+      ^$(i +(i), hstk +.hstk)
+
+
+#### Examples
+
+    > (fand [3]~ ~[1 2 3])
+    ~[2]
+
+    > (fand [4]~ ~[1 2 3])
+    ~
+
+    > (fand ['a']~ "cbabab")
+    ~[2 4]
+
+    > (fand "ba" "cbabab")
+    ~[1 3]
+
+
+
+***
+### `++find`
+
+First index in `list`
+
+Produces the index of the first occurrence of `list` `nedl` in
+`list` `hstk` as the `unit` of an atom.
+
```

Comment:
Same thing with `list` here... they look weird and nothing is lost by dropping them.